src/pages/index.js

```diff
@@ -30,36 +33,98 @@ import {
   config,
 } from "../utils/constants.js";
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-section.renderItems();
+const handleAddCardSubmit = (inputValues) => {
+  api.createCard(inputValues).then((res) => {
+    const cardElement = createCard(res);
+    section.addItem(cardElement);
+    newCardPopup.close();
+    addFormValidator.resetValidation();
+  });
+};
 const newCardPopup = new PopupWithForm("#profile-add-modal", (inputValues) => {
-  const cardElement = createCard(inputValues);
-  // cardsList.prepend(cardElement);
-  section.addItem(cardElement);
-  newCardPopup.close();
-  // addForm.reset();
-  addFormValidator.resetValidation();
+  newCardPopup.setSubmitButtonText("Saving....");
+  api
+    .createCard(inputValues)
+    .then((res) => {
+      const cardElement = createCard(res);
+      section.addItem(cardElement);
+      newCardPopup.close();
+      addFormValidator.resetValidation();
+    })
+    .finally(() => {
+      newCardPopup.setSubmitButtonText("Save");
+    });
 });
+const confirmDeleteModal = new PopupWithForm(
+  "#confirm-card-delete",
+  (inputValues) => {
+    confirmDeleteModal.setSubmitButtonText("Deleting....");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.deleteCard();
+        confirmDeleteModal.close();
+        renderItems();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmDeleteModal.setSubmitButtonText("Delete");
+      });
+  }
+);
+const changeProfilePicture = new PopupWithForm(
+  "#profile-picture-icon",
+  (inputValues) => {
+    changeProfilePicture.setSubmitButtonText("Saving....");
+    api
+      .updateProfilePicture(inputValues)
+      .then((res) => {
+        userInfo.setUserAvatar(res);
+        changeProfilePicture.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        changeProfilePicture.setSubmitButtonText("Save");
+      });
+  }
+);
+
+confirmDeleteModal.setEventListeners();
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   (inputValues) => {
-    userInfo.setUserInfo(inputValues);
     console.log(inputValues);
-    // editForm.reset();
+
     editFormValidator.resetValidation();
+
     editProfilePopup.close();
```

Comment:
Needs correction: you should close the popup and reset the validation only after the successful api response in `then` block otherwise the form will close even if an error occurred